### PR TITLE
refactor: migrate Modal to TailwindCSS

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4034]

### Description

This PR is the POC of migrating a complex component from MUI4 to TailwindCSS. We chose Modal.

This PR involves:
- Install TailwindCSS and make it work in the Storybook
- Apply BASE tokens to the TailwindCSS config
- Migrate the Modal component

### How to test

- [Temploy](https://picasso.toptal.net/fx-4034)
- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [ ] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [ ] Self reviewed
- [ ] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [ ] codemod is created and showcased in the changeset
- [ ] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4034]: https://toptal-core.atlassian.net/browse/FX-4034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "@changesets/get-github-info": "^0.5.1",
     "@cypress/skip-test": "^2.6.1",
     "@storybook/addon-a11y": "^6.5.15",
+    "@storybook/addon-styling": "0.3",
```
Comment (TomasSlama) [inline package.json:66]: we need to stick to the `0.x` version till we update to storybook 7

File: packages/picasso/src/Modal/Modal.tsx [hunk 2]
```diff
@@ -216,43 +210,50 @@ export const Modal = forwardRef<HTMLElement, Props>(function Modal(props, ref) {
     [disableBackdropClick, onBackdropClick, onClose]
   )
 
+  useBodyScrollLock(open)
```
Comment (TomasSlama) [inline packages/picasso/src/Modal/Modal.tsx:213]: even though `Modal` from mui is supporting body lock by default, it does not work when we change the container from the body to the Picasso root. It appears to be working even if we change it back to body, but I didn't want to break anything unexpectedly, so keeping the same behavior as on master

File: packages/picasso/src/Modal/Modal.tsx [hunk 3]
```diff
@@ -216,43 +210,50 @@ export const Modal = forwardRef<HTMLElement, Props>(function Modal(props, ref) {
...
+
+  const duration = transitionProps?.timeout || transitionDuration
```
Comment (TomasSlama) [inline packages/picasso/src/Modal/Modal.tsx:215]: I double checked in MUI implementation that `transitionProps` are overriding `transitionDuration`

File: packages/picasso/src/Modal/__snapshots__/test.tsx.snap [hunk 4]
```diff
@@ -10,100 +10,94 @@ exports[`Modal renders 1`] = `
...
-      style="opacity: 1; webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
+      class="opacity-0 transition-opacity MuiModal-backdrop fixed -z inset-0 bg-black bg-opacity -webkit-tap-highlight-color"
+      style="transition-duration: 300ms; opacity: 1;"
     />
     <div
-      data-test="sentinelStart"
+      data-testid="sentinelStart"
       tabindex="0"
     />
     <div
```
Comment (TomasSlama) [inline packages/picasso/src/Modal/__snapshots__/test.tsx.snap:25]: I decided to remove `MuiDialog-container` as it is not necessary and default `closeOnBackdropClick` works. If in the adopting phase, we realize that users are relying on the DOM structure, we can easily return the div back, but we will have to introduce a logic, that will close the modal if we click on this container, but not on the paper component

File: packages/picasso/src/PromptModal/PromptModal.tsx [hunk 5]
```diff
@@ -54,108 +47,103 @@ export interface Props extends Omit<ModalProps, 'children' | 'onSubmit'> {
...
-      setError(false)
-
-      await onSubmit(result)
-
-      setLoading(false)
-      handleOnAfterSubmit()
-    } catch (err) {
-      setError(true)
-      setLoading(false)
+export const PromptModal = forwardRef<HTMLDivElement, Props>(
```
Comment (TomasSlama) [inline packages/picasso/src/PromptModal/PromptModal.tsx:50]: The only change is 
```diff
- HTMLElement
+ HTMLDivElement
```
And the removal of the styles, as they were empty anyway

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

**Pull request closed without merging**